src/timestamp.rs

```diff
@@ -10,34 +11,27 @@ pub enum Timestamp {
     /// Hour, Minute and Second in UTC
     HHMMSS(u8, u8, u8),
     /// Unsupported timestamp format
-    Unsupported(String),
+    Unsupported(Vec<u8>),
 }
 
-impl FromStr for Timestamp {
-    type Err = AprsError;
-
-    fn from_str(s: &str) -> Result<Self, <Self as FromStr>::Err> {
-        let b = s.as_bytes();
+impl TryFrom<&[u8]> for Timestamp {
+    type Error = AprsError;
 
+    fn try_from(b: &[u8]) -> Result<Self, Self::Error> {
         if b.len() != 7 {
-            return Err(AprsError::InvalidTimestamp(s.to_owned()));
+            return Err(AprsError::InvalidTimestamp(b.to_owned()));
         }
 
-        let one = s[0..2]
-            .parse::<u8>()
-            .map_err(|_| AprsError::InvalidTimestamp(s.to_owned()))?;
-        let two = s[2..4]
-            .parse::<u8>()
-            .map_err(|_| AprsError::InvalidTimestamp(s.to_owned()))?;
-        let three = s[4..6]
-            .parse::<u8>()
-            .map_err(|_| AprsError::InvalidTimestamp(s.to_owned()))?;
+        let one = parse_bytes(&b[0..2]).ok_or_else(|| AprsError::InvalidTimestamp(b.to_owned()))?;
```

Comment:
This will error out early if the timestamp contains invalid unicode, even if it's a `Timestamp::Unsupported`, which doesn't seem to require valid unicode. I'm not familiar with APRS at all though, so it's totally possible I'm missing something here.

If it _is_ the case that we can assume the encoded timestamp is valid unicode, even when it's unsupported, we might as well just do `String::from_utf8` at the beginning and change the `Unsupported` variant to `Unsupported(String)`.

Reply:
The APRS standard is pretty relaxed. I think supporting non-UTF8 timestamps would be good, to be safe. I'll figure something out here